eos-updater/poll.c

```diff
@@ -470,7 +474,9 @@ check_for_update_following_checkpoint_commits (OstreeRepo     *repo,
                               error))
     return FALSE;
 
-  out_update_ref_info->refspec = g_steal_pointer (&upgrade_refspec);
+  /* The "refspec" member is the *currently booted* refspec
+   * which may get cleaned up later if we change away from it. */
+  out_update_ref_info->refspec = g_steal_pointer (&booted_refspec);
```

Comment:
So, is this the actual bug fix? Because `update_ref_info.refspec` gets passed to `eos_update_info_new` as the `old_refspec` argument?

Reply:
Yup, that's the fix. Essentially, we weren't cleaning up the booted refspec, but rather the refspec we initially considered for pulling (eg, we were measuring the difference between the refspec to initially pull vs eol-rebase refspecs, not the booted refspec vs the refspec that actually got pulled).